app/src/main/cpp/skyline/nce.cpp

```diff
@@ -122,42 +149,56 @@ namespace skyline {
     void NCE::StartThread(u64 entryArg, u32 handle, std::shared_ptr<kernel::type::KThread> &thread) {
         auto ctx = reinterpret_cast<ThreadContext *>(thread->ctxMemory->kernel.address);
         while (ctx->state != ThreadState::WaitInit);
+
         ctx->tpidrroEl0 = thread->tls;
         ctx->registers.x0 = entryArg;
         ctx->registers.x1 = handle;
         ctx->state = ThreadState::WaitRun;
-        state.logger->Debug("Starting kernel thread for guest thread: {}", thread->pid);
-        threadMap[thread->pid] = std::make_shared<std::thread>(&NCE::KernelThread, this, thread->pid);
+
+        state.logger->Debug("Starting kernel thread for guest thread: {}", thread->tid);
+        threadMap[thread->tid] = std::make_shared<std::thread>(&NCE::KernelThread, this, thread->tid);
     }
 
     void NCE::ThreadTrace(u16 numHist, ThreadContext *ctx) {
         std::string raw;
         std::string trace;
         std::string regStr;
+
         ctx = ctx ? ctx : state.ctx;
+
         if (numHist) {
             std::vector<u32> instrs(numHist);
             u64 size = sizeof(u32) * numHist;
             u64 offset = ctx->pc - size + (2 * sizeof(u32));
```

Comment:
I think offset can be auto

Reply:
I have added the following addendum to `Usage of auto`: "In addition, prefer not to use `auto` in cases where it's hard to determine the return type due to assigned value being complex."